openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -3476,6 +3483,34 @@ def __init__(self, **kwargs):
 
             super().__init__(**kwargs)
 
+            if sigma:
+                self._extra_nb_var = "rmin_half"
+            if rmin_half:
+                self._extra_nb_var = "sigma"
+
+        def __setattr__(self, name, value):
+            if name == "rmin_half":
+                super().__setattr__(name, value)
+                super().__setattr__("sigma", value / 2 ** (1 / 6))
+                self._extra_nb_var = "sigma"
+
+            if name == "sigma":
+                super().__setattr__(name, value)
+                super().__setattr__("rmin_half", value * 2 ** (1 / 6))
+                self._extra_nb_var = "rmin_half"
+
+            super().__setattr__(key=name, value=value)
```

Comment:
We can get rid of the `super.__setattr__(name, value)` lines inside the `if` blocks here, since the unconditional `__setattr__` should run regardless (I tested this change locally and it seemed fine) 

```suggestion
            super().__setattr__(key=name, value=value)
            if name == "rmin_half":
                super().__setattr__("sigma", value / 2 ** (1 / 6))
                self._extra_nb_var = "sigma"

            if name == "sigma":
                super().__setattr__("rmin_half", value * 2 ** (1 / 6))
                self._extra_nb_var = "rmin_half"

```